### PR TITLE
Export: Check status of export while in progress

### PR DESCRIPTION
**Pull request by jordwest**
This change polls the server for the status of an export after the export has begun.

### Automated Testing
```
(cd client/state/site-settings/exporter && make test)
```
The following tests are new in this PR:
```
  actions
    #exportStatusFetch()
      ✓ should dispatch fetch export status action when thunk triggered
      ✓ should dispatch export complete action when an export has completed
      ✓ should dispatch export failure action when an export has failed
  selectors
    #isExporting()
      ✓ should return false if state is not known
      ✓ should return false if an export hasn't started yet
      ✓ should return true if an export is in progress
```

### Browser Testing
1. Switch to a small test site
2. Go to **Settings > Export**
3. Click the **Export All** button. A spinner will appear and the button disables:
screen shot 2016-02-12 at 10 07 05 pm 2
4. After some time, a notice should indicate completion:
screen shot 2016-02-12 at 10 07 14 pm 2

The network tab of the developer console should show a series of polling requests for the export status:
screen_shot_2016-02-12_at_10_10_09_pm

### CODE REVIEW

**Comment by mtias** [comment]
@rickybanister I think a notice like this shouldn't have the dismiss button, or at least the dismiss should be part of the dark pill?

screen shot 2016-02-12 at 10 07 14 pm 2

**Comment by rickybanister** [comment]
Here is my design for this:

screen shot 2016-02-12 at 11 47 20 am

I don't think a global notice is appropriate here. It can be a regular notice and just let it stick around until it expires.

**Comment by rickybanister** [comment]
I do like the idea of putting the dismiss within the action part (or at least detecting if there is both an action and a dismiss and matching their background tint).

**Comment by mtias** [comment]
What happens if you want to navigate away form this page while it's exporting?

**Comment by dllh** [comment]
Yeah, this navigating away but wanting to get a notification of the completed export (which can take some time for larger sites) was the rationale for using a global notice.

Why do you have to initiate a new export if you dismiss the notice? You'll get it via email anyway. (But also: Maybe that's also a rationale for making the notice not global, since you'll get the email and an in-product notification is thus not all that important.)

**Comment by rickybanister** [comment]
I had hoped we could hook into the notifications system, you mention an 'in-product notification', will that be happening? I think if you navigated away you'd get a notification and an email with links to your file and that would be a better (more spacious) experience.

I also think the in-page notice could be different than the one we show if you navigate away (a shorter one that just takes you back to the export page?). The in-page notice doesn't need to be global, I think the one in the design works better (doesn't cover the navigation). If we do show a notice if you navigate away it could be a short global notice that says `Export Complete! | Download` with no dismiss button. Once you click a download link it would automatically dismiss the notice. We may still need to allow for a dismiss button AND and action, but I don't think it's necessary here yet.

> Why do you have to initiate a new export if you dismiss the notice?

I figured people would initiate a new export if they dismissed just out of confusion. In that moment the link to download is gone (despite it being in their email). I assume they'd go 'aw shucks' and hit the big blue Export button a second time.

**Comment by mtias** [comment]
> If we do show a notice if you navigate away it could be a short global notice that says Export Complete! | Download with no dismiss button.

:+1:

**Comment by dllh** [comment]
There aren't immediate plans to shift notification of export into the notifications system. I think it's a neat idea, but I don't want to hold up shipping this PR/feature over it. I'm fine with whatever you think makes sense UI-wise here. If the in-page notice is preferable, I'm cool with it for sure, and if it does go missing because a user navs away or whatever, they'll get the email, so we're closing the loop.

@jordwest can you switch the notice here and plan to follow up re using the notifications system for this once we've got our basic feature shipped (and have landed Support User)?

**Comment by jordwest** [comment]
I agree that local notices will be fine for a first release. I've added some commits to switch to the inline notice components and keep with the original design:

### Error state
screen shot 2016-02-16 at 9 02 39 pm

### Completed state
screen shot 2016-02-16 at 10 01 58 pm

The CircleCI test is failing due to that one unusual test that fails every now and then. The tests are all passing locally.

**Comment by mtias** [comment]
@jordwest I think we could remove the dismiss actions on both of those notices.

File: client/my-sites/exporter/exporter.jsx [hunk 1]
```diff
@@ -49,8 +55,37 @@ export default React.createClass( {
...
+		}
+		if ( this.props.didFail ) {
+			notice = (
+				<Notice
+					status="is-error"
+					text={ this.translate( 'There was a problem preparing your export file. Please check your connection and try again, or contact support.' ) }
+					onDismissClick={ dismissNotice }
+				>
+					<NoticeAction href={ '/help/contact' }>
+						Get Help
```
Comment (mtias) [inline client/my-sites/exporter/exporter.jsx]: Missing translation?

**Comment by jordwest** [comment]
@matias: I've just pushed a couple of commits to add the translations and remove the dismiss buttons:

screen shot 2016-02-17 at 9 17 46 am

**Comment by dllh** [comment]
The code here looks good to me, and I've tested to confirm functionality. Let's :ship:.

**Pull request merged**